src/pages/Search/SearchMain.js

```diff
@@ -0,0 +1,97 @@
+import React, { useState, useEffect } from 'react';
+import DefaultView from './DefaultView';
+import { KeywordView } from './KeywordView';
+import { ResultView } from './ResultView';
+
+import styled from 'styled-components';
+import { Close } from '@styled-icons/ionicons-solid/Close';
+import { Search } from '@styled-icons/bootstrap/Search';
+import theme from '../../styles/theme';
+
+const API = 'http://54.180.126.99:8000';
+
+const SearchMain = ({
+  placeholder,
+  searchTarget,
+  isOnFocus,
+  isInputFocusOn,
+  searchValue,
+  handleInputKeyword,
+  clearSearchValue,
+  isResultClicked,
+  isResultViewOn,
+}) => {
+  const [filterResult, setFilterResult] = useState([]);
+  const [totalCount, setTotalCount] = useState(0);
+
+  useEffect(() => {
+    fetch(
+      `${API}/books/search?Search_Target=${searchTarget}&target=${searchValue}`
+    )
+      .then(res => res.json())
+      .then(res => {
+        setFilterResult(res.RESULT);
+        setTotalCount(res.books_count);
+      });
+  }, [searchValue]);
+
+  return (
+    <>
+      <Wrapper>
+        <Icon onClick={isResultViewOn} />
+        <Input
+          placeholder={placeholder}
+          onChange={handleInputKeyword}
+          onFocus={isInputFocusOn}
+          value={searchValue}
+        />
+        {searchValue.length > 0 && <CloseButton onClick={clearSearchValue} />}
+      </Wrapper>
+      {!isOnFocus && searchValue.length == 0 && <DefaultView />}
+      {isOnFocus && !isResultClicked && searchValue.length > 0 && (
```

Comment:
```suggestion
      {isOnFocus && !isResultClicked && searchValue.length && (
```
- searchValue의 length가 0 아닌 값이면 이미 true로 평가 되기 때문에 생략 가능해요
- 조건이 좀 복잡한데 저 조건문들에 적당한 이름을 주어 변수로 선언할 수 있을까요?